client/knode/validator/states.go

```diff
@@ -45,22 +45,30 @@ func (val *validator) notLoggedInState() stateFn {
 		chainHeadSub := val.chain.SubscribeChainHeadEvent(chainHeadCh)
 		defer chainHeadSub.Unsubscribe()
 
-		txHash, err := val.consensus.Join(val.walletAccount, val.deposit)
+		isValidator, err := val.consensus.IsValidator(val.walletAccount.Account().Address)
 		if err != nil {
-			log.Error("Error joining validators network", "err", err)
-			return nil
+			log.Crit("Failed to verify if account is already a validator")
 		}
 
-		log.Info("Waiting confirmation to participate in the consensus")
-
-		receipt, err := tx.WaitMined(context.TODO(), val.backend, txHash)
-		if err != nil {
-			log.Crit("Failed to verify the voter registration", "err", err)
-		}
-		if receipt.Status == types.ReceiptStatusFailed {
-			log.Crit("Failed to register the validator - receipt status failed")
+		if !isValidator {
+			txHash, err := val.consensus.Join(val.walletAccount, val.deposit)
+			if err != nil {
+				log.Error("Error joining validators network", "err", err)
+				return nil
+			}
+			log.Info("Waiting confirmation to participate in the consensus")
+
+			receipt, err := tx.WaitMined(context.TODO(), val.backend, txHash)
+			if err != nil {
+				log.Crit("Failed to verify the voter registration", "err", err)
```

Comment:
Hmmm... If this check failes, a node will panic, a user will start it again, the resume mechanism will start, and it can happend that node will die here agait.
Guys, I think this can be a crash loop.

Reply:
Is there a possibility to make this in an e2e case? I dont fully understand the issue, is something that must be fixed here? Or should be an isolated issue?

@yourheropaul @rgeraldes @JekaMas

Reply:
I think it can be isolated into a separate issue. We won't have any miners that we don't control on the network for a while.